src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -661,6 +670,78 @@ private BoundExpression BindRefType(ExpressionSyntax node, DiagnosticBag diagnos
             return new BoundTypeExpression(node, null, CreateErrorType("ref"));
         }
 
+        private BoundExpression BindReturnExpression(ReturnExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            bool hasErrors = false;
+            if (!IsJumpExpressionInProperContext(node))
+            {
+                diagnostics.Add(ErrorCode.ERR_ThrowMisplaced, node.ReturnKeyword.GetLocation());
```

Comment:
Could I change `ERR_ThrowMispalced` message to `A {0} expression is not allowed in this context.` to be used for all jump expressions?